flow/scripts/report_metrics.tcl

```diff
@@ -68,3 +68,23 @@ proc report_metrics { when {include_erc true} } {
 
   puts ""
 }
+
+proc report_fmax { when } {
+  puts "report fmax at $when"
+  set time_unit [sta::unit_scale_abreviation "time"]
+  set slack_scale_factor 1
+  if {$time_unit == "p"} {
+    set slack_scale_factor 0.001
+  }
+
+  foreach clk [all_clocks] {
+    set clk_name [get_name $clk]
+    set clk_period [get_property $clk "period"]
+    report_checks -path_delay max -format full_clock_expanded -path_group $clk_name
+    set clk_slack [sta::time_sta_ui [sta::worst_slack_cmd "max"]]
```

Comment:
I assume this is the worst slack across all clocks not $clk

Reply:
@jjcherry56  -- Can you comment and provide some insight to lines 83 and 84.  Ideally I would like to extract the fmax for each of the clock domains.

Reply:
There isn't much that is right about this from the very first line to the last so I would just throw out the PR and ask how to do it right. It is completely confused about units, there is no reason to call 'report_checks' and the accessor for slack with respect to a clock is wrong.